compiler/src/linking/link.re

```diff
@@ -578,24 +608,26 @@ let link_all = (linked_mod, dependencies, signature) => {
   if (Grain_utils.Config.use_start_section^) {
     Function.set_start(linked_mod, start);
   } else {
-    ignore @@ Export.add_function_export(linked_mod, start_name, grain_start);
+    ignore @@
+    Export.add_function_export(linked_mod, start_name, Compcore.grain_start);
   };
 };
 
 let link_modules = ({asm: wasm_mod, signature}) => {
-  G.clear(dependency_graph);
   Hashtbl.clear(modules);
+  Hashtbl.clear(hard_dependencies);
 
   let main_module = Module_resolution.current_filename^();
   Hashtbl.add(modules, main_module, wasm_mod);
+  Hashtbl.add(hard_dependencies, main_module, ());
 
-  G.add_vertex(dependency_graph, main_module);
-  build_dependency_graph(
+  load_hard_dependencies(
     ~base_dir=Filepath.String.dirname(main_module),
     main_module,
   );
-  let dependencies =
-    Topo.fold((dep, acc) => [dep, ...acc], dependency_graph, []);
+
+  let dependencies = Module_resolution.get_dependencies();
```

Comment:
This gives us the full dependency graph, which includes files that might not contain necessary code.